src/senselab/audio/tasks/speaker_diarization/api.py

```diff
@@ -4,12 +4,12 @@
 
 from senselab.audio.data_structures import Audio
 from senselab.audio.tasks.speaker_diarization.pyannote import diarize_audios_with_pyannote
-from senselab.utils.data_structures import DeviceType, PyannoteAudioModel, ScriptLine, SenselabModel
+from senselab.utils.data_structures import DeviceType, PyannoteAudioModel, ScriptLine
 
 
 def diarize_audios(
     audios: List[Audio],
-    model: SenselabModel = PyannoteAudioModel(path_or_uri="pyannote/speaker-diarization-3.1", revision="main"),
+    model: Optional[PyannoteAudioModel] = None,
```

Comment:
I would use SenselabModel as an input type here, since in the general api we don't want only PyannoteAudioModel objects. I know it's consistent with the current status of implementation, but doesn't communicate properly the intention of the method. And it's same for all other api methods.

Reply:
In my opinion, for the static type hints we should use what the function actually supports, not what it would in the future. Because these are explicitly checked by mypy, it seems less confusing to align the type hinting to the actual type check that occurs within the function. And then when we implement other models we can change the type hint. I see your point though and I can make it clearer in the docstring and errors that we would like to support more models in the future.